src/modules/cpu/kernel/transpose_generic.hpp

```diff
@@ -0,0 +1,482 @@
+{
+    if (nDim == 0)
+    {
+        *dst = *src;
+    }
+    else
+    {
+        for (int i = 0; i < *dstShape; i++)
+        {
+            transpose(dst, dstStrides + 1, src, srcStrides + 1, dstShape + 1, nDim - 1);
+            dst += *dstStrides;
+            src += *srcStrides;
+        }
+    }
+}
+
+RppStatus transpose_generic_f32_f32_host_tensor(Rpp32f *srcPtr,
+                                                RpptGenericDescPtr srcGenericDescPtr,
+                                                Rpp32f *dstPtr,
+                                                RpptGenericDescPtr dstGenericDescPtr,
+                                                Rpp32u *permTensor,
+                                                Rpp32u *roiTensor,
+                                                RppLayoutParams layoutParams,
+                                                rpp::Handle& handle)
+{
+    Rpp32u numThreads = handle.GetNumThreads();
+    Rpp32u nDim = dstGenericDescPtr->numDims;
+    Rpp32u batchSize = dstGenericDescPtr->dims[0];
+
+    // allocate buffer for input strides, output strides, output shapes
+    Rpp32u *srcStridesTensor = (Rpp32u *)calloc(nDim * batchSize, sizeof(int));
+    Rpp32u *dstStridesTensor = (Rpp32u *)calloc(nDim * batchSize, sizeof(int));
+    Rpp32u *dstShapeTensor = (Rpp32u *)calloc(nDim * batchSize, sizeof(int));
+
+    omp_set_dynamic(0);
+#pragma omp parallel for num_threads(numThreads)
+    for(int batchCount = 0; batchCount < batchSize; batchCount++)
+    {
+        Rpp32f *srcPtrTemp, *dstPtrTemp;
+        srcPtrTemp = srcPtr + batchCount * srcGenericDescPtr->strides[0];
+        dstPtrTemp = dstPtr + batchCount * dstGenericDescPtr->strides[0];
+
+        Rpp32u *dstShape = dstShapeTensor + batchCount * nDim;
+        Rpp32u *roi = roiTensor + batchCount * nDim * 2;
+        Rpp32u *begin = roi;
+        Rpp32u *length = &roi[nDim];
+        Rpp32u *perm = permTensor;
+        Rpp32u *srcStrides = srcStridesTensor + batchCount * nDim;
+        Rpp32u *dstStrides = dstStridesTensor + batchCount * nDim;
+
+        bool copyInput = true;
+        for(int i = 0; i < nDim; i++)
+            copyInput *= (perm[i] == i);
+
+        // do memcpy of input to output since output order is same as input order
+        if(copyInput)
+        {
+            memcpy(dstPtrTemp, srcPtrTemp, (size_t)(srcGenericDescPtr->strides[0] * sizeof(Rpp32f)));
+        }
+        else
+        {
+            for(int i = 1; i < nDim; i++)
+                srcPtrTemp += begin[i] * srcGenericDescPtr->strides[i];
+
+            if (nDim == 2 && perm[0] == 1 && perm[1] == 0)
+            {
+                compute_2d_transpose(srcPtrTemp, dstPtrTemp, length[0], length[1], srcGenericDescPtr->strides[1], dstGenericDescPtr->strides[1]);
+            }
+            else if (nDim == 3)
+            {
+                if(perm[0] == 2 && perm[1] == 0 && perm[2] == 1 && length[2] == 16)
```

Comment:
Like compute_2d_transpose(), could you consolidate as compute_3d_transpose()?

Reply:
For 2D inputs, there is only 1 transpose order possible and it can be used in 3D transpose
So i consolidated to a single function in case of 2D. But for 3D inputs there are more than 1 transpose order possible, so i could not consolidate transpose orders for 3D into single function since it becomes too long